geo/resolver.go

```diff
@@ -0,0 +1,198 @@
+package geo
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"time"
+
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"googlemaps.github.io/maps"
+
+	"github.com/bitmark-inc/autonomy-api/schema"
+	"github.com/bitmark-inc/autonomy-api/utils"
+)
+
+var (
+	ErrNoGeoInfoFound         = fmt.Errorf("no geo information found")
+	ErrResolverNotInitialized = fmt.Errorf("location resolver is not initialized")
+)
+
+var (
+	Taiwan = "Taiwan"
+	US     = "United States"
+)
```

Comment:
We support Iceland. Please add Iceland

Reply:
I can added it, but it is not used in any places for now. Do you insist to add it?

Reply:
I think there is only Taiwan now. This makes me confused. For example, Taiwan does not use CDC data anymore. The code takes care of it. Does US have a special rule of it? Why do you say that Iceland  does not have a special rule? For consistency,  we should code it in order to let people know we need to take care of the cases  country by country

Reply:
What I said **special rules** might be misleading here. What I mean is that I need to convert field in a certain way. We have country, state and county in the database. Whereas, we do not have that from google map api. They use `administrative_area_level_1`, `administrative_area_level_2`. We need map them into specific field when needed instead of list all the countries. Most of them could map directly.

Reply:
As my understanding of your words, I summarized the rule is 
Check on the Country we supported. 
- If it is a country level, we don't need to take care of it so we do not add cases
- If the level is below the country, we need to specify how to map state and county
- Taiwan data is based on Taiwan CDC data, it should be deprecated

Am I correct with the logic in your mind?

Reply:
No. This is a package to resolve coordinates to location. Give a coordinate, I resolve the details. Originally, I preserves the county converting for Taiwan just want to make alignment with the originally code. I remove it now.

Reply:
> If the level is below the country, we need to specify how to map state and county

Only US have this for now.

Reply:
As I understand this is the resolver that resolves geo-position (lat, long) to political location which is (country, state, county, city).  Every country has her own naming of political location and that is why geo API naming its the level below country admin_1 ... etc. We have Taiwan, United States, and Iceland so far. Only United States uses the level below country that is why it needs covert state=admin-1.  Am I wrong this? Which part is wrong ?

Reply:
> No. This is a package to resolve coordinates to location. Give a coordinate, I resolve the details. Originally, I preserves the county converting for Taiwan just want to make alignment with the originally code. I remove it now.

I think this what it is. The rules I list is based on this. Can you give me the rule otherwise I don't know what to do when next time I add a new Country.

Reply:
I think there is nothing wrong. I just list some example results from google here.

```
Country: "United States", administrative_area_level_1: "California", administrative_area_level_2: "San Diego County"
Country: "Taiwan", administrative_area_level_1: "Taipei", administrative_area_level_2: ""
Country: "Taiwan", administrative_area_level_1: "", administrative_area_level_2: "Yilan County"
Country: "Iceland", administrative_area_level_1: "Capital Region", administrative_area_level_2: ""
```

Reply:
> Can you give me the rule otherwise I don't know what to do when next time I add a new Country.

The rule come from google. When you find there is a country does not parsed well, you need to query it and see why it is different from others.